ant.js

```diff
@@ -0,0 +1,161 @@
+var xoff = 0;
+var yoff = 0;
+
+function Ant(x, y, nest) {
+  this.pos = createVector(x, y);
+  this.prevPos = this.pos.copy();
+  this.vel = createVector(0, 0);
+  this.acc = createVector(0,0);
+  this.maxspeed = 1;
+  this.maxforce = 0.2;
+  this.d = 5;
+  this.inc = 0.1;
+
+  this.nestPos = createVector(nest.pos.x, nest.pos.y);
+  this.nestRadius = nest.d / 2;
+
+  this.hasFood = false;
+  this.timeGotFood = null;
+
+  this.updatePrev = function() {
+    this.prevPos.x = this.pos.x;
+    this.prevPos.y = this.pos.y;
+  };
+
+  this.run = function(ants) {
+    this.coordinate();
+    this.borders();
+    this.foodExpire();
+    this.render();
+    this.updatePrev();
+  };
+
+  this.coordinate = function() {
```

Comment:
I like the way you've encapsulated most of the logic within well-named functions here which makes this pretty advanced coordination logic easier to read through.